enrichments/trace/internal/elastic/span.go

```diff
@@ -237,6 +238,39 @@ func (s *spanEnrichmentContext) enrichTransaction(
 	}
 }
 
+// In OTel a root span can represent an outgoing call or a producer span
+// In such cases, the span is still mapped into a transaction, but such spans are enriched
+// with additional attributes that are specific to the outgoing call or producer span.
+func (s *spanEnrichmentContext) enrichExitSpanTransaction(
+	span ptrace.Span,
+	cfg config.Config,
+) {
+	if span.Kind() == ptrace.SpanKindClient || span.Kind() == ptrace.SpanKindProducer {
+		if cfg.Span.TypeSubtype.Enabled {
+			s.setSpanTypeSubtype(span)
+		}
+		if cfg.Span.ServiceTarget.Enabled {
```

Comment:
@gregkalapos What about the thing we discussed about having two `ProcessorEvent` values in this case?

Reply:
I tested it and that triggers an error when the water-flow chart is generated. These transactions will be also returned for all queries that look for spans and there were some missing span related fields there.

@axw had a related idea: we could just remove the filter on `processor.event` from the relevant queries in Kibana. I'd rather explore that option and potentially get rid of `processor.event` in those queries.

Reply:
Discussed this with @AlexanderWert and we said, the correct way would be to still set both `span` and `transaction` and it'd be nice to not have to update Kibana.

In https://github.com/elastic/opentelemetry-lib/pull/125/commits/3def749edba2df55722e7deaba60e0d0521c3821 I changed this.

Now that we share the `enrichSpan()` method, there are no missing field errors when the waterfall chart is generated. With above commit, this seems working:

<img width="2263" alt="Screenshot 2024-12-09 at 19 28 32" src="https://github.com/user-attachments/assets/9781dfa3-bb4a-485f-a686-79f1dcd55c58">

However, order matters here. [This is the code in Kibana](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers.ts#L299-L308) that builds up the chart - and if it's `[span transaction]`, then it's still categorized as a span and not as a transaction leading to an empty waterfall chart.

So, it seems we can set both `span` and `transaction` on such root exit spans, but I'd say mid/long term we should still update Kibana to make it more robust.